### PR TITLE
Users/pingke/posw

### PR DESCRIPTION
**Pull request by ping-ke**

### CODE REVIEW

File: cluster/config/config.go [hunk 1]
```diff
@@ -80,15 +88,63 @@ func NewPOSWConfig() *POSWConfig {
...
+	}
+}
+
+func (c *RootPOSWConfig) GetEnabled() bool                { return c.Enabled }
+func (c *RootPOSWConfig) GetEnableTimestamp() uint64      { return c.EnableTimestamp }
+func (c *RootPOSWConfig) GetWindowSize() uint64           { return c.WindowSize }
+func (c *RootPOSWConfig) GetTotalStakePerBlock() *big.Int { return c.TotalStakePerBlock }
+
+func (c *RootPOSWConfig) GetDiffDivider(blocktime uint64) uint64 {
+	diffDriver := c.DiffDivider
```
Comment (qizhou) [inline cluster/config/config.go:129]: diffDriver => diffDivider?

File: mainnet/singularity/cluster_config_template.json [hunk 2]
```diff
@@ -84,12 +84,16 @@
...
         "ENABLED": true,
         "ENABLE_TIMESTAMP": 1569567600,
         "DIFF_DIVIDER": 10000,
         "WINDOW_SIZE": 512,
-        "TOTAL_STAKE_PER_BLOCK": 1000000000000000000000000
+        "TOTAL_STAKE_PER_BLOCK": 1000000000000000000000000,
+        "BOOST_TIMESTAMP": 1646064000,
+        "BOOST_MULTIPLER_PER_STEP": 2,
+        "BOOST_STEPS": 8,
+        "BOOST_SETP_INTERVAL": 172800
```
Comment (qizhou) [inline mainnet/singularity/cluster_config_template.json:96]: Types: SETP => STEP

File: devnet/cluster_config_template.json [hunk 3]
```diff
@@ -52,7 +52,16 @@
...
+      "ROOT_POSW_CONFIG": {
+        "ENABLED": true,
+        "ENABLE_TIMESTAMP": 1645200000,
+        "DIFF_DIVIDER": 100,
+        "WINDOW_SIZE": 512,
+        "TOTAL_STAKE_PER_BLOCK": 10000000000000000000000,
+        "BOOST_TIMESTAMP": 1645372800,
+        "BOOST_MULTIPLER_PER_STEP": 2,
+        "BOOST_STEPS": 8,
+        "BOOST_SETP_INTERVAL": 86400
```
Comment (qizhou) [inline devnet/cluster_config_template.json:64]: Type: SETP => STEP

File: cluster/config/config.go [hunk 4]
```diff
@@ -80,15 +88,63 @@ func NewPOSWConfig() *POSWConfig {
...
+type RootPOSWConfig struct {
+	Enabled               bool     `json:"ENABLED"`
+	EnableTimestamp       uint64   `json:"ENABLE_TIMESTAMP"`
+	DiffDivider           uint64   `json:"DIFF_DIVIDER"`
+	WindowSize            uint64   `json:"WINDOW_SIZE"`
+	TotalStakePerBlock    *big.Int `json:"TOTAL_STAKE_PER_BLOCK"`
+	BoostTimestamp        uint64   `json:"BOOST_TIMESTAMP"`
+	BoostMultiplerPerStep uint64   `json:"BOOST_MULTIPLER_PER_STEP"`
+	BoostSteps            uint64   `json:"BOOST_STEPS"`
+	BoostStepInterval     uint64   `json:"BOOST_SETP_INTERVAL"`
```
Comment (qizhou) [inline cluster/config/config.go:106]: Type: SETP => STEP

**Comment by ping-ke** [comment]
replace by https://github.com/QuarkChain/goquarkchain/pull/650.

**Pull request closed without merging**